update(readme): Swap dynamoDb for SSM (#19)
coveo-labs/autoreply-coveo-slack-bot: README.md

## Components
* __Coveo platform__ - hosts the index and organization for your data, provides search capabilities.
* __Coveo Impersonation__ - key for the authentication against the platform (currently only fetching public documents based on the query pipeline).
* __Slack Admin__ access to add a new Application.
* __AWS Lambda__ - runs your NodeJs code that will interact with the Coveo platform without provisioning or managing infrastructure.

> * __AWS Parameter Store__- stores the search token cache (the app token needs consist of: Slack App token, Slack Bot token and the Slack Signing Secret).

Author: ```suggestion
* __AWS Parameter Store__- stores the search token cache (the app token needs consist of: The Coveo APi Key,  Slack App token, Slack Bot token and the Slack Signing Secret).
```

* __AWS API Gateway__, - points to the above AWS Lambda function.

## How it works

> In Slack, in a channel where the bot will have been previously added, users asking a question will receive an answer based on the top Coveo Search answers. The trigger in this project is a question mark, but that can be changed easily changed.
> 

Author: ```suggestion
In Slack, in a channel where the bot will have been previously added, users asking a question will receive an answer based on the top Coveo Search answers. The trigger in this project is a question mark, but that can be easily changed.

```

> Slack will send the new message event to the AWS Lambda code, which will only trigger and respond if there is a question mark in the text field, and if so, will send a query to Coveo based on the keywords in the said text. The top 3 results are returned to Slack.

Author: ```suggestion
Slack will send the new message event to the AWS Lambda code, which will only trigger and respond if there is a question mark in the text field, and if so, will send a query to Coveo based on the keywords in the said text. The top 3 results will then be returned to Slack.
```

   **_Keep that key private!!!_**

   Insert it only in the AWS Parameter Store  

#### Create the Parameter Store Folder for your App 
  * Navigate to the Parameter Store landing page, and click on `Create parameter`
  ![image](https://user-images.githubusercontent.com/73175206/169589442-634442dc-bcf2-4297-a1bb-6aea7ac0a8a1.png)

>   * Create a new name path for your application.For this example, we used the following path `/rd/coveo-autoreply-bot` with the parameter name `COVEO_API_KEY`. Make sure to give a unique path to your app related tokens as we will need to give restricted acces to those token later on in the configuration. Also make sure to select the `SecureString` type for the parameter. 

Author: ```suggestion
  * Create a new name path for your application.For this example, we used the following path `/rd/coveo-autoreply-bot` with the parameter name `COVEO_API_KEY`. Make sure to give a unique path to your app-related tokens as we will need to give our Lambda restricted access to it later in the configuration. Also, make sure to select the `SecureString` type for the parameter. 
```
